g3doc/Tutorials/Toy/Ch-2.md

```diff
@@ -8,52 +8,67 @@ to compile Toy.
 ## Introduction: Multi-Level IR
 
 Other compilers like LLVM (see the
-[Kaleidoscope tutorial](https://llvm.org/docs/tutorial/LangImpl01.html)) offer
-a fixed set of predefined types and, usually *low-level* / RISC-like,
+[Kaleidoscope tutorial](https://llvm.org/docs/tutorial/MyFirstLanguageFrontend/index.html))
+offer a fixed set of predefined types and, usually *low-level* / RISC-like,
 instructions. It is up to the frontend for a given language to perform any
 language specific type-checking, analysis, or transformation before emitting
 LLVM IR. For example, clang will use its AST to perform static analysis but also
-transformation like C++ template instantiation through AST cloning and rewrite.
-Finally, languages with construction higher-level than C/C++ may require
+transformations like C++ template instantiation through AST cloning and rewrite.
+Finally, languages with construction at a higher-level than C/C++ may require
 non-trivial lowering from their AST to generate LLVM IR.
 
 As a consequence, multiple frontends end up reimplementing significant pieces of
 infrastructure to support the need for these analyses and transformation. MLIR
-addresses this issue by being designed for extensibility. As such, there is
-little to no pre-defined set of instructions (*operations* in MLIR
-terminology) or types.
-
-## MLIR Module, Functions, Blocks, and Operations
-
-[Language reference](../../LangRef.md#operations)
-
-In MLIR (like in LLVM), the top level structure for the IR is a Module
-(equivalent to a translation unit in C/C++). A module contains a list of
-functions, and each function has a list of blocks forming a CFG. Each block is a
-list of operations that execute in sequence.
-
-Operations in MLIR are similar to instructions in LLVM, however MLIR does not
-have a closed set of operations. Instead, MLIR operations are fully extensible
-and can have application-specific semantics.
+addresses this issue by being designed for extensibility. As such, there are
+little to no pre-defined instructions (*operations* in MLIR terminology) or
+types.
+
+## MLIR Dialects, Regions, Blocks, and Operations
+
+[Language reference](../../LangRef.md#dialects)
+
+In MLIR, the core unit of abstraction and computation is an `Operation`, similar
+in many ways to LLVM instructions. Operations may also hold a number
+[`Region`s](../../LangRef.md#regions), of which contain a list of
+[`Blocks`](../../LangRef.md#blocks) forming a CFG(Control Flow Graph). `Blocks`
+are similar to LLVM `BasicBlocks` in that they contain a sequential list of
+`Operations`. Operations can be used to represent all of the core IR structures
+in LLVM: instructions, globals(like functions), modules, etc; however MLIR does
+not have a closed set of operations. Instead, the MLIR operation set is fully
```

Comment:
Should we mention standard ops and types available in mlir (such as tensor) here?

Reply:
Standard ops here isn't important, but mentioning that we will reuse the standard tensor type is.